### ISSUE TITLE
0.15.6 java.lang.ClassCastException

### ISSUE THREAD

**Issue by Batbold-Gansukh** [tminglei/slick-pg#385]
After upgrading to 0.15.6, it throws exception while executing query if a table contains an array type column.

```
java.lang.ClassCastException: com.zaxxer.hikari.pool.HikariProxyConnection cannot be cast to org.postgresql.jdbc.PgConnection[error] c.BonusController - com.zaxxer.hikari.pool.HikariProxyConnection cannot be cast to org.postgresql.jdbc.PgConnection
java.lang.ClassCastException: com.zaxxer.hikari.pool.HikariProxyConnection cannot be cast to org.postgresql.jdbc.PgConnection
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.mkArray(PgArrayJdbcTypes.scala:41)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.setValue(PgArrayJdbcTypes.scala:31)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.setValue(PgArrayJdbcTypes.scala:13)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$WrappedConvArrayJdbcType.setValue(PgArrayJdbcTypes.scala:106)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$WrappedConvArrayJdbcType.setValue(PgArrayJdbcTypes.scala:97)
	at slick.jdbc.JdbcStatementBuilderComponent$QueryBuilder.$anonfun$expr$2(JdbcStatementBuilderComponent.scala:314)
	at slick.jdbc.JdbcStatementBuilderComponent$QueryBuilder.$anonfun$expr$2$adapted(JdbcStatementBuilderComponent.scala:312)
	at slick.util.SQLBuilder.$anonfun$build$2(SQLBuilder.scala:39)
	at slick.util.SQLBuilder.$anonfun$build$2$adapted(SQLBuilder.scala:38)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:59)

	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.mkArray(PgArrayJdbcTypes.scala:41)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.setValue(PgArrayJdbcTypes.scala:31)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.setValue(PgArrayJdbcTypes.scala:13)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$WrappedConvArrayJdbcType.setValue(PgArrayJdbcTypes.scala:106)
	at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$WrappedConvArrayJdbcType.setValue(PgArrayJdbcTypes.scala:97)
	at slick.jdbc.JdbcStatementBuilderComponent$QueryBuilder.$anonfun$expr$2(JdbcStatementBuilderComponent.scala:314)
	at slick.jdbc.JdbcStatementBuilderComponent$QueryBuilder.$anonfun$expr$2$adapted(JdbcStatementBuilderComponent.scala:312)
	at slick.util.SQLBuilder.$anonfun$build$2(SQLBuilder.scala:39)
	at slick.util.SQLBuilder.$anonfun$build$2$adapted(SQLBuilder.scala:38)
	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:52)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at slick.util.SQLBuilder.$anonfun$build$1(SQLBuilder.scala:38)
	at slick.util.SQLBuilder.$anonfun$build$1$adapted(SQLBuilder.scala:36)
	at slick.jdbc.JdbcInvokerComponent$QueryInvokerImpl.setParam(JdbcInvokerComponent.scala:35)
	at slick.jdbc.StatementInvoker.results(StatementInvoker.scala:33)
	at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:21)
	at slick.jdbc.Invoker.foreach(Invoker.scala:47)
	at slick.jdbc.Invoker.foreach$(Invoker.scala:46)
	at slick.jdbc.StatementInvoker.foreach(StatementInvoker.scala:15)
	at slick.jdbc.StreamingInvokerAction.run(StreamingInvokerAction.scala:22)
	at slick.jdbc.StreamingInvokerAction.run$(StreamingInvokerAction.scala:20)
	at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$1.run(JdbcActionComponent.scala:215)
	at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$1.run(JdbcActionComponent.scala:215)
	at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:242)
	at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:242)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by tminglei** [comment]
Well, my fault, I forget the connection pool.

**Comment by tminglei** [comment]
Hi @Batbold-Gansukh fixed in `v0.15.7`.
Thanks! 👍

**Comment by pantelissd** [comment]
Hi @tminglei I am having a similar issue with version 0.17.0 and slick 3.2.3

java.lang.ClassCastException: com.sun.proxy.$Proxy2 cannot be cast to org.postgresql.jdbc.PgConnection
        at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.mkArray(PgArrayJdbcTypes.scala:42)
        at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.setValue(PgArrayJdbcTypes.scala:31)
        at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$SimpleArrayJdbcType.setValue(PgArrayJdbcTypes.scala:13)
        at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$WrappedConvArrayJdbcType.setValue(PgArrayJdbcTypes.scala:107)
        at com.github.tminglei.slickpg.array.PgArrayJdbcTypes$WrappedConvArrayJdbcType.setValue(PgArrayJdbcTypes.scala:98)
        at slick.jdbc.JdbcStatementBuilderComponent$QueryBuilder.$anonfun$expr$14(JdbcStatementBuilderComponent.scala:408)
        at slick.jdbc.JdbcStatementBuilderComponent$QueryBuilder.$anonfun$expr$14$adapted(JdbcStatementBuilderComponent.scala:406)
        at slick.util.SQLBuilder.$anonfun$build$2(SQLBuilder.scala:39)
        at slick.util.SQLBuilder.$anonfun$build$2$adapted(SQLBuilder.scala:38)
        at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:58)
        at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:51)
        at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
        at slick.util.SQLBuilder.$anonfun$build$1(SQLBuilder.scala:38)
        at slick.util.SQLBuilder.$anonfun$build$1$adapted(SQLBuilder.scala:36)
        at slick.jdbc.JdbcInvokerComponent$QueryInvokerImpl.setParam(JdbcInvokerComponent.scala:35)
        at slick.jdbc.StatementInvoker.results(StatementInvoker.scala:33)
        at slick.jdbc.StatementInvoker.iteratorTo(StatementInvoker.scala:21)
        at slick.jdbc.Invoker.foreach(Invoker.scala:47)
        at slick.jdbc.Invoker.foreach$(Invoker.scala:46)
        at slick.jdbc.StatementInvoker.foreach(StatementInvoker.scala:15)
        at slick.jdbc.StreamingInvokerAction.run(StreamingInvokerAction.scala:22)
        at slick.jdbc.StreamingInvokerAction.run$(StreamingInvokerAction.scala:20)
        at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$1.run(JdbcActionComponent.scala:216)
        at slick.jdbc.JdbcActionComponent$QueryActionExtensionMethodsImpl$$anon$1.run(JdbcActionComponent.scala:216)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:275)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:275)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
